internal/gardener/shoot/hyperscaler/aws/provider_specific_config.go

```diff
@@ -2,12 +2,43 @@ package aws
 
 import (
 	"encoding/json"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func GetInfrastructureConfig(_ string, _ []string) ([]byte, error) {
-	return json.Marshal("{}")
+const infrastructureConfigKind = "InfrastructureConfig"
+const controlPlaneConfigKind = "ControlPlaneConfig"
+const apiVersion = "aws.provider.extensions.gardener.cloud/v1alpha1"
+
+func GetInfrastructureConfig(workersCidr string, zones []string) ([]byte, error) {
+	return json.Marshal(NewInfrastructureConfig(workersCidr, zones))
 }
 
 func GetControlPlaneConfig() ([]byte, error) {
-	return json.Marshal("{}")
+	return json.Marshal(NewControlPlaneConfig())
+}
+
+func NewInfrastructureConfig(workersCidr string, zones []string) InfrastructureConfig {
+	return InfrastructureConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       infrastructureConfigKind,
+			APIVersion: apiVersion,
+		},
+		Networks: Networks{
+			Zones: generateAWSZones(workersCidr, zones),
+			VPC: VPC{
+				CIDR: &workersCidr,
+			},
+		},
+	}
+}
+
+func NewControlPlaneConfig() *ControlPlaneConfig {
+	return &ControlPlaneConfig{
+		TypeMeta: v1.TypeMeta{
+			Kind:       controlPlaneConfigKind,
+			APIVersion: apiVersion,
+		},
+	}
```

Comment:
We are missing AWS WorkerConfig configuration. See the function [func (c AWSGardenerConfig) ExtendShootConfig](https://github.com/kyma-project/control-plane/blob/ea75326a0e2f1108f4297dd04b8090cd75266898/components/provisioner/internal/model/gardener_config.go#L632) in Provisioner repository. 


There are three elements configured: 

1. InfrastructureConfig
2. ControlPlaneConfig
3. Workers with AWS specific provider config taking into account parameter `EnableIMDSv2` [added recently](https://github.com/kyma-project/control-plane/pull/3429) (I'm not sure if could be missing in Runtime CR)

Code from provisioner with AWS specific worker configuration: 

```golang
	if c.input.EnableIMDSv2 != nil && *c.input.EnableIMDSv2 {
		awsWorkerConfig := NewAWSWorkerConfig(awsIMDSv2HTTPPutResponseHopLimit)
		jsonWCData, err := json.Marshal(awsWorkerConfig)
		if err != nil {
			return apperrors.Internal("error encoding aws worker config: %s", err.Error())
		}
		workers[0].ProviderConfig = &apimachineryRuntime.RawExtension{Raw: jsonWCData}
	}
```

Reply:
Done